src/components/typeahead.jsx

```diff
@@ -49,6 +50,9 @@ export default React.createClass({
         listOpen: nextProps.list.length !== 0
       });
     }
+    if (nextProps.val && nextProps.manualMode) {
```

Comment:
Since both `if` statements in `componentWillReceiveProps` check for `nextProps.manualMode`, let's wrap that check in an outer statement:
```
componentWillReceiveProps(nextProps) {
  if (nextProps.manualMode) {
    if (nextProps.list) {
      ...
    }
    if (nextProps.val) {
      ...
    }
  }
}
```